cscs-checks/system/io/ior_check.py

```diff
@@ -1,160 +1,165 @@
+        'ior_block_size': '48g',
+        'fulen': {
+            'num_tasks': 8,
+            'build_system_cc': 'mpicc',
+            'build_system_cxx': 'mpic++',
+            'valid_prog_environs': ['PrgEnv-gnu']
+        }
+    }
+}
+
+# Setting some default values
+for data in fs.values():
+    if 'ior_block_size' not in data:
+        data['ior_block_size'] = '24g'
+    if 'ior_access_type' not in data:
+        data['ior_access_type'] = 'MPIIO'
+    if 'reference' not in data:
+        data['reference'] = {
+            'read_bw': (1, -0.5, None),
+            'write_bw': (1, -0.5, None)
+        }
+
 
 class IorCheck(rfm.RegressionTest):
-    def __init__(self, fs_mount_point):
+    def __init__(self, base_dir):
         super().__init__()
-        self.descr = 'IOR check (%s)' % fs_mount_point
-        self.tags = {'ops', fs_mount_point}
-
-        if fs_mount_point == '/scratch/snx1600':
-            self.valid_systems = ['daint:gpu']
-            self.num_tasks = 2400
-            self.num_tasks_per_node = 12
-        elif fs_mount_point == '/scratch/snx1600tds':
-            self.valid_systems = ['dom:gpu']
-            self.num_tasks = 192
-            self.num_tasks_per_node = 12
-        elif fs_mount_point == '/scratch/snx2000':
-            self.valid_systems = ['dom:gpu']
-            if self.current_system.name == 'dom':
-                self.num_tasks = 192
-                self.num_tasks_per_node = 12
-            else:
-                self.num_tasks = 1
-                self.num_tasks_per_node = 1
-        elif fs_mount_point == '/scratch/snx2000tds':
-            self.valid_systems = ['dom:gpu']
-            if self.current_system.name == 'dom':
-                self.num_tasks = 192
-                self.num_tasks_per_node = 12
-            else:
-                self.num_tasks = 1
-                self.num_tasks_per_node = 1
-        elif fs_mount_point == '/scratch/snx3000':
-            self.valid_systems = ['daint:gpu']
-            self.num_tasks = 720
-            self.num_tasks_per_node = 12
-        elif fs_mount_point == '/users':
-            self.valid_systems = ['daint:gpu', 'dom:gpu']
-            self.num_tasks = 1
-            self.num_tasks_per_node = 1
-            self.tags |= {'maintenance'}
-        elif fs_mount_point == '/apps':
-            self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.descr = 'IOR check (%s)' % base_dir
+        self.tags = {'ops', base_dir}
+
+        self.base_dir = base_dir
+        self.username = getpass.getuser()
+        self.test_dir = os.path.join(self.base_dir, self.username, '.ior')
+        self.test_file = os.path.join(self.test_dir, 'ior.dat')
+
+        try:
+            os.mkdir(self.test_dir)
+        except OSError:
+            pass
+
+        self.valid_systems = fs[base_dir]['valid_systems']
+
+        cur_sys = self.current_system.name
+
+        try:
+            self.num_tasks = fs[base_dir][cur_sys]['num_tasks']
+        except KeyError:
```

Comment:
Also instead of catching `KeyError` and using a default value, you can use the dictionary's `get()` method, e.g.,:

```python
self.num_tasks = fs[base_dir][cur_sys].get('num_tasks', 1)
```

This will still throw `KeyError` if `cur_sys` is not found. Generally, though, you should handle the case where `cur_sys` is unknown, since somebody might try to run your test on an unknown system.

Reply:
What's the best way to abort the test in case cur_sys is not in the self.fs dict?